pkg/monitor/monitor/util.go

```diff
@@ -837,8 +837,7 @@ func getMonitorStatefulSetSkeleton(sa *core.ServiceAccount, monitor *v1alpha1.Ti
 			},
 			Template: core.PodTemplateSpec{
 				ObjectMeta: meta.ObjectMeta{
-					Labels:      buildTidbMonitorLabel(monitor.Name),
-					Annotations: monitor.Spec.Annotations,
```

Comment:
No need to remove this, right?

Reply:
The `LastAppliedPodTemplate` in L809 can be removed, please help confirm.

Reply:
We should not use the `monitor.Spec.Annotations` directly here and L827, because if we modify the annotation laster, the `monitor.Spec.Annotations` will be modified too.

Reply:
We also should get a deepcopy of the tidbmonitor before the sync to avoid it being changed during the sync.

Reply:
ok.I know.

Reply:
We should also set annotation for the Pods, right?

Reply:
What about https://github.com/pingcap/tidb-operator/pull/3570/files#r537363895 this comment?